diff --git a/include/configs/apalis-imx8.h b/include/configs/apalis-imx8.h
index 2ff591a822..9843d65a65 100644
--- a/include/configs/apalis-imx8.h
+++ b/include/configs/apalis-imx8.h
@@ -148,11 +148,10 @@
 #define CONFIG_SYS_MEMTEST_END		0x89000000
 
 /* Environment in eMMC, before config block at the end of 1st "boot sector" */
-#define CONFIG_ENV_SIZE			0x2000
-#define CONFIG_ENV_OFFSET		(-CONFIG_ENV_SIZE + \
-					 CONFIG_TDX_CFG_BLOCK_OFFSET)
-#define CONFIG_SYS_MMC_ENV_DEV		0
-#define CONFIG_SYS_MMC_ENV_PART		1
+#define CONFIG_ENV_SIZE			 0x2000
+#define CONFIG_ENV_OFFSET		 0x800000
+#define CONFIG_ENV_OFFSET_REDUND	 0x1000000
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT 1
 
 #define CONFIG_SYS_MMC_IMG_LOAD_PART	1
 
